tests/integration/conftest.py

```diff
@@ -271,9 +281,13 @@ def fail_on_errors():
 
 
 @pytest.fixture
-def s3():
+def s3(pytestconfig):
     if not AWS_ACCESS_KEY_ID or not AWS_SECRET_ACCESS_KEY:
-        pytest.skip("No AWS credentials")
+        msg = "No AWS credentials"
+        if pytestconfig.getoption("ci"):
+            pytest.fail(msg)
```

Comment:
In theory this kind of stuff could be monitored by measuring coverage on the testsuite itself. Then you can find tests that are improperly skipped or not discovered at all for whatever reason.

In practice it's much harder to set up.

Reply:
I think that could work if you have a 100% test coverage enforced.  But anything less and it probably doesn't.  I like one global that you can use to potentially skip things on local workstations.

Reply:
I mean testing coverage of the code in `tests/`, not the code that is being tested. Surely all tests are running in CI?